### DISCUSSION TITLE
Problems with vertically aligning to middle?

### DISCUSSION THREAD

**Discussion by RealXuChe** [lvjr/tabularray#251] [Q&A]
I'm trying to put some code highlighted with `minted` in one cell, but I found it's hard to align to the middle vertically.
```latex
\documentclass{article}

\usepackage{xcolor}
\usepackage{minted}
\usepackage{tabularray}

\begin{document}

\begin{tblr}{|Q[m,c]|Q[c,m]|}
\hline
\fbox{\mintinline{python}{a=2}} & \fbox{(1)} \\
\hline
\end{tblr}

\end{document}
```
Result on my computer is:
![image](https://user-images.githubusercontent.com/47659370/169745795-5b01a84b-9205-4b33-872b-aca8ef4f9f34.png)
It's obvious that the spacing above the code block is smaller than the one below.
Could anyone please help me middling this?

**Comment by lvjr** [comment]
There is no difference between vertical alignment `t`/`m`/`b` if every cell has only a single line. You can use [`adjustbox`](https://ctan.org/pkg/adjustbox) package to adjust the cells:
```latex
\documentclass{article}

\usepackage{xcolor}
\usepackage{tabularray}
\usepackage{adjustbox}

\begin{document}

\begin{tblr}{|Q[c]|Q[c]|}
\hline
\adjustbox{fbox,valign=m}{a=2} & \adjustbox{fbox,valign=m}{(1)} \\
\hline
\end{tblr}

\end{document}
```

![image](https://user-images.githubusercontent.com/4239949/169764717-6b6ce201-d464-44c5-9a5b-ae72839e9ace.png)

**Reply by RealXuChe** [reply]
Thanks for helping! This seems to work fine for the case, so I tried to play with `adjustbox` more. I just copied your solution three times and changed `fboxsep` for each of them, and some weird thing happened:
```latex
\documentclass{article}

\usepackage{xcolor}
\usepackage{tabularray}
\usepackage{adjustbox}

\begin{document}

\begin{tblr}{|Q[c]|Q[c]|}
\hline
\adjustbox{fbox=0.4pt 0pt,valign=m}{\tt a=2} & \adjustbox{fbox,valign=m}{(1)} \\
\hline
\end{tblr}

\begin{tblr}{|Q[c]|Q[c]|}
\hline
\adjustbox{fbox,valign=m}{\tt a=2} & \adjustbox{fbox=0.4pt 0pt,valign=m}{(1)} \\
\hline
\end{tblr}

\begin{tblr}{|Q[c]|Q[c]|}
\hline
\adjustbox{fbox=0.4pt 0pt,valign=m}{\tt a=2} & \adjustbox{fbox=0.4pt 0pt,valign=m}{(1)} \\
\hline
\end{tblr}

\end{document}
```
If I only change the first column, everything's fine. When I set `fboxsep` of the second column, both columns will break. Setting both `fboxsep` will only make it worse.
![image](https://user-images.githubusercontent.com/47659370/169851839-988c10f3-117d-4413-b210-a5eb1eb4ab3c.png)
I'm basically new to LaTeX. It's confusing for me that the order of the changed fbox will affect the result. I wonder why this would happen and how to fix it.

**Reply by lvjr** [reply]
By default, there is an invisible strut in every cell to give minimal height and depth for it (see `\strut` command in the answer of Herbert Voß). You can remove the struts by setting `stretch=0`.

```latex
\usepackage{xcolor}
\usepackage{tabularray}
\usepackage{adjustbox}

\begin{document}

\begin{tblr}{colspec={|Q[c]|Q[c]|},stretch=0}
\hline
\adjustbox{fbox=0.4pt 0pt,valign=m}{\tt a=2} & \adjustbox{fbox,valign=m}{(1)} \\
\hline
\end{tblr}

\begin{tblr}{colspec={|Q[c]|Q[c]|},stretch=0}
\hline
\adjustbox{fbox,valign=m}{\tt a=2} & \adjustbox{fbox=0.4pt 0pt,valign=m}{(1)} \\
\hline
\end{tblr}

\begin{tblr}{colspec={|Q[c]|Q[c]|},stretch=0}
\hline
\adjustbox{fbox=0.4pt 0pt,valign=m}{\tt a=2} & \adjustbox{fbox=0.4pt 0pt,valign=m}{(1)} \\
\hline
\end{tblr}

\end{document}
```

![image](https://user-images.githubusercontent.com/4239949/169925356-95eb86cc-bcf9-4a6d-855e-7534bad58f32.png)

**Comment by hvoss49** [comment]
```
\documentclass{article}
\usepackage{tabularray}
\begin{document}

\begin{tblr}{|Q[c]|Q[c]|}\hline
\fbox{a=2\strut} & \fbox{(1)\strut} \\\hline
\end{tblr}

\end{document}
```
<img width="180" alt="Bildschirmfoto 2022-05-23 um 11 02 46" src="https://user-images.githubusercontent.com/12891199/169784197-043fb06b-f42d-4ad8-a94c-7262af1e35b3.png">

**Comment by RealXuChe** [comment]
I think I'd better have a deeper look at the TeX mechanism and try to figure it out. And thanks for all your help!
